### PR TITLE
begin adding record flags struct

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
NONE
```

Begins wiring RecordFlags through the apply command.

cc @soltysh @deads2k

### CODE REVIEW

**Comment by deads2k** [comment]
@kubernetes/sig-cli-maintainers

File: pkg/kubectl/cmd/apply.go [hunk 1]
```diff
@@ -207,14 +222,26 @@ func parsePruneResources(mapper meta.RESTMapper, gvks []string) ([]pruneResource
 	return pruneResources, nil
 }
 
-func RunApply(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, options *ApplyOptions) error {
+func (o *ApplyOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
+	o.RecordFlags.Complete(f.Command(cmd, false))
+	recorder, err := o.RecordFlags.ToRecorder()
```
Comment (deads2k) [inline pkg/kubectl/cmd/apply.go]:
```go
var err error
o.Recorder, err = o.RecordFlags.ToRecorder()
```

File: pkg/kubectl/cmd/apply.go [hunk 2]
```diff
@@ -271,12 +298,8 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, opti
 			visitedNamespaces.Insert(info.Namespace)
 		}
 
-		// Add change-cause annotation to resource info if it should be recorded
-		if cmdutil.ShouldRecord(cmd, info) {
-			recordInObj := info.Object
-			if err := cmdutil.RecordChangeCause(recordInObj, f.Command(cmd, false)); err != nil {
-				glog.V(4).Infof("error recording current command: %v", err)
-			}
+		if err := o.Recorder.Record(info.Object); err != nil {
```
Comment (deads2k) [inline pkg/kubectl/cmd/apply.go]: much better

**Comment by deads2k** [comment]
nit, then lgtm.  Sort out your red though.

/approve

**Review by soltysh — APPROVED** [review]
/lgtm

**Review by soltysh — APPROVED** [review]
/lgtm

**Pull request merged**